httpcore/_async/connection_pool.py

```diff
@@ -326,6 +326,10 @@ async def aclose(self) -> None:
             self._requests = []
 
     async def __aenter__(self) -> "AsyncConnectionPool":
+        # Acquiring the pool lock here ensures that we have the
+        # correct dependencies installed as early as possible.
+        async with self._pool_lock:
+            pass
```

Comment:
I think here don't need extra await with immediately lock release.

```suggestion
        # Ensure correct dependencies installed as early as possible.
        if not self._pool_lock._backend:
            self._pool_lock.setup()
```

Reply:
I'd generally prefer using the lock's public API here.